docroot/modules/custom/nvli_custom_search_api/src/Plugin/rest/resource/NvliHomepageListResource.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+/**
+ * @file
+ * Contains Drupal\nvli_custom_search_api\Plugin\rest\resource.
+ */
+
+namespace Drupal\nvli_custom_search_api\Plugin\rest\resource;
+
+use Drupal\Core\Session\AccountProxyInterface;
+use Drupal\rest\Plugin\ResourceBase;
+use Drupal\rest\ResourceResponse;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException;
+use Symfony\Component\HttpKernel\Exception\HttpException;
+use Psr\Log\LoggerInterface;
+use Drupal\Core\Entity;
+use Drupal\custom_solr_search\SearchSolrAll;
+use Drupal\custom_solr_search\Search;
+use Symfony\Component\HttpFoundation;
+
+/**
+ * Provides a resource to get view modes by entity and bundle.
+ *
+ * @RestResource(
+ *   id = "custom_rest_homepage_listing_resource",
+ *   label = @Translation("Custom rest homepage listing resource"),
+ *   uri_paths = {
+ *     "canonical" = "/rest/v1/homepage/listing"
```

Comment:
API Endpoints should be named based on the resource its pulling. Definitely not page based.